presto-main/src/main/java/com/facebook/presto/metadata/FunctionListBuilder.java

```diff
@@ -149,9 +159,65 @@ public FunctionListBuilder scalar(Class<?> clazz)
         catch (IllegalAccessException e) {
             throw Throwables.propagate(e);
         }
+        groupMatchingScalars();
+        return this;
+    }
+
+    private FunctionListBuilder groupMatchingScalars()
+    {
+        List<ParametricFunction> newFunctions = groupMatchingScalars(functions);
+        functions.clear();
+        functions.addAll(newFunctions);
         return this;
     }
 
+    private List<ParametricFunction> groupMatchingScalars(List<ParametricFunction> inputFunctions)
+    {
+        inputFunctions.forEach(f -> checkArgument(f instanceof FunctionInfo, "expected FunctionInfo but got %s", f));
+        Map<Signature, List<FunctionInfo>> groupedFunctions = inputFunctions.stream()
+                .map(f -> (FunctionInfo) f)
+                .collect(Collectors.groupingBy(ParametricFunction::getSignature));
```

Comment:
how would it work if I write `@Param("DECIMAL(x + y)")` and `@Param("DECIMAL(x   (maaaaany spaces)                + y)")`?